src/main/java/net/jonathangiles/tools/teenyhttpd/json/TeenyJson.java

```diff
@@ -0,0 +1,352 @@
+package net.jonathangiles.tools.teenyhttpd.json;
+
+
+import net.jonathangiles.tools.teenyhttpd.implementation.ReflectionUtils;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * 0 dependencies Teeny JSON serializer
+ *
+ */
+public class TeenyJson {
+
+    private final JsonEncoder encoder;
+    private final Map<Class<?>, ValueParser<?>> parsers = new ConcurrentHashMap<>();
+
+    public TeenyJson() {
+        encoder = new JsonEncoder();
+    }
+
+    public String writeValueAsString(Object value) {
+        return encoder.writeValueAsString(value);
+    }
+
+    public void writeValue(BufferedOutputStream baos, Object value) throws IOException {
+        baos.write(writeValueAsString(value).getBytes());
+    }
+
+    /**
+     * Read a JSON string and return a collection of objects of the given type.
+     *
+     * @param collectionType one of List or Set
+     * @param json the JSON string to parse
+     * @return the parsed object which could be null.
+     */
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    public <T, K> K readCollection(String json, Class<? extends Collection> collectionType, Class<T> type) {
```

Comment:
Shouldn't the second parameter be `Class<K extends Collection> collectionType`?

Reply:
Yes, but this may be the best choice:

```java
public <T, K extends Collection> K readCollection(String json, Class<K> collectionType, Class<T> type)
```

Reply:
Had to switch back to 
```java
public <T, K> K readCollection(String json, Class<? extends Collection> collectionType, Class<T> type) {
```

because this:

```java
public <T, K extends Collection> K readCollection(String json, Class<K> collectionType, Class<T> type)
```

Raises a warning about `unchecked assignment` 😪


<img width="690" alt="Screenshot 2024-03-07 at 10 01 43 p m" src="https://github.com/JonathanGiles/TeenyHttpd/assets/25180512/ccfc6f4d-dcac-4b20-a919-718f2f1b265b">